src/runtime/description.ts

```diff
@@ -23,17 +23,31 @@ import {Dictionary} from './hot.js';
 
 export class Description {
   private constructor(
-      private readonly storeDescById: Dictionary<string> = {},
-      // TODO(mmandlis): replace Particle[] with serializable json objects.
-      private readonly arcRecipes: {patterns: string[], particles: Particle[]}[],
-      private readonly particleDescriptions: ParticleDescription[] = []) {
+    private readonly storeDescById: Dictionary<string> = {},
+    // TODO(mmandlis): replace Particle[] with serializable json objects.
+    private readonly arcRecipes: {patterns: string[], particles: Particle[]}[],
+    private readonly particleDescriptions: ParticleDescription[] = []) {
   }
 
-  static async createForPlan(plan: Recipe): Promise<Description> {
+  static async XcreateForPlan(plan: Recipe): Promise<Description> {
     const particleDescriptions = await Description.initDescriptionHandles(plan.particles);
     return new Description({}, [{patterns: plan.patterns, particles: plan.particles}], particleDescriptions);
   }
 
+  static async createForPlan(arc: Arc, plan: Recipe): Promise<Description> {
```

Comment:
optional - maybe we should just have a single method `static async create(arc, plan, relevance?) that works for both cases (it may fallback to activeRecipe, if no plan was passed), WDYT?